internal/codespaces/api/api_test.go

```diff
@@ -126,13 +131,99 @@ func createFakeCreateEndpointServer(t *testing.T, wantStatus int) *httptest.Serv
 	}))
 }
 
+func createHttpClient() (*http.Client, error) {
+	return &http.Client{}, nil
+}
+
+func TestNewApiServerUrl_dotcomConfig(t *testing.T) {
+	os.Setenv("GITHUB_API_URL", "")
```

Comment:
```suggestion
	t.Setenv("GITHUB_API_URL", "")
```

Let's make sure to use `t.Setenv` in tests rather than `os.Setenv`.

Reply:
Thanks for pointing this out! I wasn't aware of this method, but I agree it's much better for tests after looking it up. I actually modeled this after a few others tests that were calling `os.Setenv`. Would you like me to update all of them in this PR?

https://github.com/search?q=repo%3Acli%2Fcli+os.setEnv+path%3A**%2F*_test.go&type=code

Reply:
I went ahead and updated them all and kept the non-codespaces ones in another commit so that it's easy to rebase it out if you want to keep that separate.